pkg/controller/projectclaim/customresourceadapter.go

```diff
@@ -168,3 +169,54 @@ func (c *CustomResourceAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimS
 	}
 	return nil
 }
+
+// SetProjectClaimCondition sets a condition on a ProjectClaim resource's status
+func (c *CustomResourceAdapter) SetProjectClaimCondition(
+	status corev1.ConditionStatus,
+	reason string,
+	message string,
+) []gcpv1alpha1.ProjectClaimCondition {
+	conditions := c.projectClaim.Status.Conditions
+	conditionType := gcpv1alpha1.ClaimConditionError
+	now := metav1.Now()
+	existingCondition := c.FindProjectClaimCondition()
+	if existingCondition == nil {
+		if status == corev1.ConditionTrue {
+			conditions = append(
+				conditions,
+				gcpv1alpha1.ProjectClaimCondition{
+					Type:               conditionType,
+					Status:             status,
+					Reason:             reason,
+					Message:            message,
+					LastTransitionTime: now,
+					LastProbeTime:      now,
+				},
+			)
+		}
+	} else {
+		// If it does not exist, assign it as now. Otherwise, do not touch
+		if existingCondition.Status != status {
+			existingCondition.LastTransitionTime = now
+		}
+		existingCondition.Status = status
+		existingCondition.Reason = reason
+		existingCondition.Message = message
+		existingCondition.LastProbeTime = now
+	}
+	return conditions
+}
+
+// FindProjectClaimCondition finds in the condition that has the
+// specified condition type in the given list. If none exists, then returns nil.
+func (c *CustomResourceAdapter) FindProjectClaimCondition() *gcpv1alpha1.ProjectClaimCondition {
```

Comment:
Shouldn't the type be passed as an argument? At least that's how I read the comment above.

Reply:
Now, we don't have to pass it. it currently uses the actual adapter's fields.

Reply:
Now I don't understand the comment anymore :D

Reply:
well, me too :D